app/routing/DeparturesRouter.scala

```diff
@@ -47,27 +47,35 @@ class DeparturesRouter @Inject() (
     with VersionedRouting {
 
   def submitDeclaration(): Action[Source[ByteString, _]] = route {
-    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) => v2Departures.submitDeclaration()
-    case _                                                    => v1Departures.submitDeclaration()
+    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON) => v2Departures.submitDeclaration()
+    case _                                                         => v1Departures.submitDeclaration()
   }
 
-  def getMessage(departureId: String, messageId: String): Action[Source[ByteString, _]] = route {
-    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) =>
-      runIfBound[V2DepartureId](
-        "departureId",
-        departureId,
-        boundDepartureId => runIfBound[V2MessageId]("messageId", messageId, v2Departures.getMessage(boundDepartureId, _))
-      )
-    case _ =>
-      runIfBound[V1DepartureId](
-        "departureId",
-        departureId,
-        boundDepartureId => runIfBound[V1MessageId]("messageId", messageId, v1DepartureMessages.getDepartureMessage(boundDepartureId, _))
-      )
-  }
+  def isValidVersionTwoAcceptHeader(headerValue: String) = """^application\/vnd\.hmrc\.2\.0\+.+$""".r.pattern.matcher(headerValue).matches
+
+  def getMessage(departureId: String, messageId: String): Action[Source[ByteString, _]] =
+    route {
+      case Some(acceptHeaderValue) if isValidVersionTwoAcceptHeader(acceptHeaderValue) =>
```

Comment:
Don't need you to change this, becauase this is perfectly fine, but you can bake the pattern match into the case statement like this.

```scala
val VERSION_2_ACCEPT_HEADER_PATTERN = """^application\/vnd\.hmrc\.2\.0\+.+$""".r

def getMessage(departureId: String, messageId: String): Action[Source[ByteString, _]] =
    route {
        case Some(VERSION_2_ACCEPT_HEADER_PATTERN()) => // v2
        case _ => // v1
    }
```

It's stylistic, but something to consider for the future.